pkg/controllers/hub/trafficmanagerbackend/controller.go

```diff
@@ -557,9 +569,9 @@ func equalAzureTrafficManagerEndpoint(current, desired armtrafficmanager.Endpoin
 		*current.Properties.EndpointStatus == *desired.Properties.EndpointStatus
 }
 
-// updateTrafficManagerEndpointsAndUpdateStatusIfUnknown updates the Azure Traffic Manager endpoints.
+// updateTrafficManagerEndpointsAndUpdateStatus updates the Azure Traffic Manager endpoints.
 // Returns the accepted endpoints and a list of bad endpoints error when it fails to create/update endpoint or not because of bad request.
-func (r *Reconciler) updateTrafficManagerEndpointsAndUpdateStatusIfUnknown(ctx context.Context, backend *fleetnetv1beta1.TrafficManagerBackend, profile *armtrafficmanager.Profile, desiredEndpoints map[string]desiredEndpoint) ([]fleetnetv1beta1.TrafficManagerEndpointStatus, []error, error) {
+func (r *Reconciler) updateTrafficManagerEndpointsAndUpdateStatus(ctx context.Context, backend *fleetnetv1beta1.TrafficManagerBackend, profile *armtrafficmanager.Profile, desiredEndpoints map[string]desiredEndpoint) ([]fleetnetv1beta1.TrafficManagerEndpointStatus, []error, error) {
 	backendKObj := klog.KObj(backend)
 	acceptedEndpoints := make([]fleetnetv1beta1.TrafficManagerEndpointStatus, 0, len(desiredEndpoints))
```

Comment:
for the accepted endpoints, we need to populate the weight to the "fromCluster" from export,

```
// FromCluster contains service configuration mapped to a specific source cluster.
type FromCluster struct {
	// ClusterStatus describes the source cluster status.
	ClusterStatus `json:",inline"`

	// Weight defines the weight configured in the serviceExport from the source cluster.
	// Possible values are from 0 to 1000.
	// +optional
	Weight *int64 `json:"weight,omitempty"`
}
```

Reply:
What's the difference between this and the Weight in the status?

Reply:
Where was this weight set before? I can't seem to find it

Reply:
I don't think this field was set before